.tools/validation/metadata_test.py

```diff
@@ -0,0 +1,378 @@
+                                    snippet_tags=[
+                                        "php.snippet.tag.1",
+                                        "php.snippet.tag.2",
+                                    ],
+                                    snippet_files=[],
+                                )
+                            ],
+                            add_services={},
+                            more_info=[],
+                        )
+                    ],
+                ),
+            ],
+            services={"sns": {}, "sqs": {}},
+        )
+    ]
+
+
+@pytest.mark.parametrize(
+    "filename,expected_errors",
+    [
+        (
+            "empty_metadata.yaml",
+            [
+                metadata_errors.MissingField(
+                    field="title",
+                    file="empty_metadata.yaml",
+                    id="sns_EmptyExample",
+                ),
+                metadata_errors.MissingField(
+                    field="title_abbrev",
+                    file="empty_metadata.yaml",
+                    id="sns_EmptyExample",
+                ),
+                metadata_errors.MissingField(
+                    field="languages",
+                    file="empty_metadata.yaml",
+                    id="sns_EmptyExample",
+                ),
+            ],
+        ),
+        (
+            "errors_metadata.yaml",
+            [
+                metadata_errors.UnknownLanguage(
+                    language="Perl",
+                    file="errors_metadata.yaml",
+                    id="sqs_WrongServiceSlug",
+                ),
+                metadata_errors.InvalidSdkGuideStart(
+                    file="errors_metadata.yaml",
+                    id="sqs_WrongServiceSlug",
+                    language="Perl",
+                    guide="https://docs.aws.amazon.com/absolute/link-to-my-guide",
+                ),
+                metadata_errors.MissingBlockContentAndExcerpt(
+                    file="errors_metadata.yaml",
+                    id="sqs_WrongServiceSlug",
+                    language="Perl",
+                    sdk_version=None,
+                ),
+                metadata_errors.MissingField(
+                    field="versions",
+                    file="errors_metadata.yaml",
+                    id="sqs_TestExample",
+                    language=None,
+                ),
+                metadata_errors.MissingBlockContentAndExcerpt(
+                    file="errors_metadata.yaml",
+                    id="sns_TestExample",
+                    language="Java",
+                    sdk_version=None,
+                ),
+                # example_errors.MissingSnippetTag(
+                #     file="errors_metadata.yaml",
+                #     id="sqs_TestExample",
+                #     language="Java",
+                #     sdk_version=2,
+                #     tag="this.snippet.does.not.exist",
+                # ),
```

Comment:
On purpose?

Reply:
Correct, we don't check snippets on the Github side (yet!)